### PR TITLE
Move validation that player owns the moving piece

### PR DESCRIPTION
**Pull request by swcraig**
One of the checks for a valid move is that the moving player owns the piece that they attempt to move. They also must not own the piece that occupies the destination cell (if there is one).

This PR adds these two basic checks to `Move#valid?`.

### CODE REVIEW

File: spec/move_spec.rb [hunk 1]
```diff
@@ -14,4 +15,44 @@
       expect(subject).to be_a Move
     end
   end
+
+  describe 'valid?' do
+    context 'an invalid move' do
+      context 'when given a cell off of the board' do
+        let(:destination_cell) { Cell.new(1, -1) }
+        subject do
```
Comment (gmacdougall) [inline spec/move_spec.rb]: This subject can be moved in to the describe box above with a little refactoring. it doesn't vary much between each test.

File: spec/move_spec.rb [hunk 2]
```diff
@@ -14,4 +15,44 @@
...
+          Move.new(board, player_one, source_cell, destination_cell).valid?
+        end
+
+        it 'returns false' do
+          expect(subject).to be false
```
Comment (gmacdougall) [inline spec/move_spec.rb]: RSpec provides some shorthand for dealing with this.  If you find your test description and definition are the same, you should consider eliminating the description.

```Ruby
it { is_expected.to be false }
```

File: lib/move.rb [hunk 3]
```diff
@@ -9,6 +9,12 @@ def initialize(board, player, source_location, destination_location)
     movement_vector_and_distance if source_cell_has_piece? && @destination_cell
   end
 
+  def valid?
+    return false unless @movement_vector && @distance
+    return false unless player_owns_piece?(@source_cell.content)
+    return false if player_owns_piece?(@destination_cell.content)
```
Comment (gmacdougall) [inline lib/move.rb]: Would all your specs pass if I changed the implementation of this method to:

```Ruby
def valid?
  false
end
```

File: lib/piece_set.rb [hunk 4]
```diff
@@ -33,4 +33,8 @@ def capital_row(player)
   def pawn_row(player)
     Array.new(SIDE_LENGTH) { Pawn.new(player) }
   end
+
+  def include?(piece)
```
Comment (gregdaynes) [inline lib/piece_set.rb]: Please rename method to something not built in

Comment (gregdaynes) [inline lib/piece_set.rb]: Discussed IRL - Someone more Ruby versed please chime in

Comment (swcraig) [inline lib/piece_set.rb]: For anyone who is looking into this, I have added `PieceSet#include?` to check whether a player's pieces (both capitals and pawns) includes a piece. `PieceSet` is not an `Array` and so does not respond to `#include?` in general.  By defining a `#include?` method on `PieceSet` someone perusing the code at the point of a call to this method may then (incorrectly) assume that a player's pieces (their `PieceSet`) is an Array (by seeing something like `player.pieces.include?`

Should this method be renamed? Is this in general a poor choice to make?

Comment (swcraig) [inline lib/piece_set.rb]: Decided to go a different direction anyway. This is no longer relevant.

**Pull request merged**